src/main/java/com/flipkart/ranger/finder/BaseServiceFinderBuilder.java

```diff
@@ -16,19 +16,20 @@
 
 package com.flipkart.ranger.finder;
 
-import com.flipkart.ranger.model.Deserializer;
-import com.flipkart.ranger.model.ServiceNodeSelector;
-import com.flipkart.ranger.model.ServiceRegistry;
-import com.flipkart.ranger.model.ShardSelector;
+import com.flipkart.ranger.model.*;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
 import org.apache.curator.framework.CuratorFramework;
 import org.apache.curator.framework.CuratorFrameworkFactory;
 import org.apache.curator.retry.ExponentialBackoffRetry;
 
-public abstract class BaseServiceFinderBuilder<T, RegistryType extends ServiceRegistry<T>, FinderType extends ServiceFinder<T, RegistryType>> {
+public abstract class BaseServiceFinderBuilder<T, RegistryType extends AbstractServiceRegistry<T>, FinderType extends ServiceFinder<T, RegistryType>> {
     private String namespace;
     private String serviceName;
     private CuratorFramework curatorFramework;
+    private String host;
```

Comment:
Keep only `curatorFramework` and `curatorSourceConfig` and `httpSourceConfig`
and simplify the builders to `withCuratorSourceConfig()` and `withHttpSourceConfig()`